packages/errors/src/message-formatter.ts

```diff
@@ -28,7 +28,7 @@ export function getHumanReadableErrorMessage<TErrorCode extends SolanaErrorCode>
             const variableName = messageFormatString.slice(state[START_INDEX] + 1, endIndex);
 
             fragments.push(
-                variableName in context ? `${context[variableName as keyof typeof context]}` : `$${variableName}`,
+                variableName in context ? `${context[variableName] as string}` : `$${variableName}`,
```

Comment:
I'm not sure this is true though, the value at `context[variableName]` might not be a string, as it is not here:

https://github.com/solana-labs/solana-web3.js/blob/master/packages/errors/src/context.ts#L493-L495

I think this is an appropriate place to suppress lint again. The suppression expresses ‘yeah, we know some of these will come up `[object Object]` but that's OK.’

Reply:
Thanks for the clarification, I didn't realize that use case. I rolled the changes and I just added the suppression.